### ISSUE TITLE
Getting 403 with Golang package but works fine with node

### ISSUE THREAD

**Issue by White2001Offl** [Danny-Dasilva/CycleTLS#57]
### Description

Hi, I came across with a weird thing. The spoof works fine with node js but when I try the same in Golang package I get 403. Not sure why.
I can send you the code for this "IF needed"
but this is so strange

### Issue Type

Bug

### Operating System

Mac OS

### Node Version

Node 16.x

### Golang Version

Go 16.x

**Comment by 89z** [comment]
Please provide example request

**Comment by White2001Offl** [comment]
Node JS

`const tokenResponse = await cycleTLS(tokenURL, {
                timeout: 10,
                body: JSON.stringify(hash),
                ja3:  data.ja3,
                userAgent: data.ua,
                proxy: LUMINATI_PROXY,
                headers: {
                    "sec-ch-ua": "\" Not A;Brand\";v=\"99\", \"Chromium\";v=\"96\", \"Google Chrome\";v=\"96\"",
                    'Accept': 'application/json, text/plain, */*',
                    'Accept-Language': 'en-US',
                    'Accept-Encoding': 'gzip, deflate, br',
                    'Host': 'Host',
                    'Origin': origin,
                    "sec-ch-ua-mobile": "?0",
                    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.45 Safari/537.36",
                    "sec-ch-ua-platform": "\"Windows\"",
                    'Referer': referrer,
                    'Content-Length': '100',
                    'Connection': 'keep-alive',
                    'Content-Type': 'application/json',
                    'x-app-name': appName,
                    'x-session-trace-id': uid1,
                    'x-app-url': appUrl,
                    'apikey': apiKey,
                    'x-request-trace-id': uid2,
                    'slsserver': slsServer,
                    "Sec-Fetch-Site": "cross-site",
                    "Sec-Fetch-Mode": "cors",
                    "Sec-Fetch-Dest": "empty",
                },
            }, 'post');`
 
 
 GO
 `res , err = client.Do(tokenURL, cycletls.Options{
			Ja3:       ja3Data,
			UserAgent: uaData,
			Proxy:     proxy,
			Timeout:   30,
			Body:      stringBody,
			Headers: map[string]string{
				"sec-ch-ua": "\" Not A;Brand\";v=\"99\", \"Chromium\";v=\"96\", \"Google Chrome\";v=\"96\"",
				"Accept": "application/json, text/plain, */*",
				"Accept-Language": "en-US",
				"Accept-Encoding": "gzip, deflate, br",
				"Host": "Host",
				"Origin": origin,
				"sec-ch-ua-mobile": "?0",
				"User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.45 Safari/537.36",
				"sec-ch-ua-platform": "\"Windows\"",
				"Referer": referrer,
				"Content-Length": "100",
				"Connection": "keep-alive",
				"Content-Type": "application/json",
				"x-app-name": appName,
				"x-session-trace-id": hash.UID1,
				"x-app-url": appUrl,
				"apikey": apiKey,
				"x-request-trace-id": hash.UID2,
				"slsserver": slsServer,
				"Sec-Fetch-Site": "cross-site",
				"Sec-Fetch-Mode": "cors",
				"Sec-Fetch-Dest": "empty",
			},
		}, "POST")`

**Comment by 89z** [comment]
Sorry but we need the full, complete request with no omissions. That includes full URL as well as full body.

**Comment by White2001Offl** [comment]
@89z if you don't mind
can I send you email or can send you through discord or telegram
I don't want that url get indexed in google or to be visible to public

**Comment by 89z** [comment]
Please run the following command and comment back with the output:

~~~
age -a -r age1zkz9g0kld97vs826j0r7p3ghvz3zcyd4tcqr2vrxpt0uux7mkycsl77hy6 request.txt
~~~

https://github.com/FiloSottile/age

**Comment by Danny-Dasilva** [comment]
> @89z if you don't mind can I send you email or can send you through discord or telegram I don't want that url get indexed in google or to be visible to public

My email is on my [profile](https://github.com/Danny-Dasilva), feel free to send me a message and I can debug

**Comment by White2001Offl** [comment]
@Danny-Dasilva Sent a mail to you

**Comment by Danny-Dasilva** [comment]
Just a public update. The latest available version of cycleTLS from npm (0.0.15-alpha.0) has a lot of features not included in the main repo. Those being ordered request headers (via fhttp), a fix for GREASE headers as well as gzip/brotli/deflate parsing. Header order most likely being the reason requests are failing on the golang side. I will do a release shortly that should resolve this.

**Comment by Danny-Dasilva** [comment]
this should be addressed by [this release](https://github.com/Danny-Dasilva/CycleTLS/pull/59). The `fhttp-caching` branch is what the 0.0.15 alpha release was based on.
